sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/implementation/ContainerRegistryRedirectPolicy.java

```diff
@@ -29,136 +28,131 @@
  */
 public final class ContainerRegistryRedirectPolicy implements HttpPipelinePolicy {
     private static final ClientLogger LOGGER = new ClientLogger(ContainerRegistryRedirectPolicy.class);
-    private static final int MAX_REDIRECT_ATTEMPTS;
-    private static final String REDIRECT_LOCATION_HEADER_NAME;
-    private static final int PERMANENT_REDIRECT_STATUS_CODE;
-    private static final int TEMPORARY_REDIRECT_STATUS_CODE;
-    private static final Set<HttpMethod> REDIRECT_ALLOWED_METHODS;
-    private static final String AUTHORIZATION;
-
-    static {
-        REDIRECT_ALLOWED_METHODS = new HashSet<>(Arrays.asList(HttpMethod.GET, HttpMethod.HEAD));
-        PERMANENT_REDIRECT_STATUS_CODE = 308;
-        TEMPORARY_REDIRECT_STATUS_CODE = 307;
-        REDIRECT_LOCATION_HEADER_NAME = "Location";
-        MAX_REDIRECT_ATTEMPTS = 3;
-        AUTHORIZATION = "Authorization";
-    }
+    private static final int MAX_REDIRECT_ATTEMPTS = 3;
+    private static final int PERMANENT_REDIRECT_STATUS_CODE = 308;
+    private static final int TEMPORARY_REDIRECT_STATUS_CODE = 307;
 
     @Override
     public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
-        return this.attemptRedirect(context, next, context.getHttpRequest(), 1, new HashSet<>());
+        context.setHttpRequest(context.getHttpRequest().copy());
+        return next.clone().process()
+            .flatMap(httpResponse -> {
+                if (!isRedirectResponse(httpResponse)) {
+                    return Mono.just(httpResponse);
+                }
+
+                return attemptRedirect(context, next, httpResponse, 1, new HashSet<>());
+            });
     }
 
     @Override
     public HttpResponse processSync(HttpPipelineCallContext context, HttpPipelineNextSyncPolicy next) {
-        return this.attemptRedirectSync(context, next, context.getHttpRequest(), 1, new HashSet<>());
+        context.setHttpRequest(context.getHttpRequest().copy());
+        HttpResponse httpResponse = next.clone().processSync();
+        if (!isRedirectResponse(httpResponse)) {
+            return httpResponse;
+        }
+
+        return attemptRedirectSync(context, next, httpResponse, 1, new HashSet<>());
+    }
+
+    private static HttpResponse mapResponse(HttpResponse oldResponse, HttpResponse newResponse) {
+        String digest = getDigestFromHeader(oldResponse.getHeaders());
+        if (digest != null) {
+            newResponse.getHeaders().set(DOCKER_DIGEST_HEADER_NAME, digest);
```

Comment:
Is `DOCKER_DIGEST_HEADER_NAME` an `HttpHeaderName`?

Reply:
yep!